app/services/common_platform/hearing_results_filter.rb

```diff
@@ -0,0 +1,20 @@
+module CommonPlatform
+  class HearingResultsFilter < ApplicationService
+    def initialize(hearing_results_body, defendant_id: nil)
+      @body = hearing_results_body.deep_stringify_keys
+      @defendant_id = defendant_id
+    end
+
+    def call
+      return @body unless @defendant_id
+
+      @body["hearing"]["prosecutionCases"] = @body["hearing"]["prosecutionCases"].map do |prosecution_case|
```

Comment:
I think we'll also need to extend the filtering to the `court_applications`?

Reply:
so maybe not. In that case,
`LaaReference.find_by(defendant_id: defendant.id, linked: true)` should filter for the only defendant that is linked.
see:
https://github.com/ministryofjustice/laa-court-data-adaptor/blob/acc9e845a59ddae97dac6a9bbeaebd0e005c146f/app/services/hearings_creator.rb#L27